packages/hardhat-core/package.json

```
@@ -23,7 +23,7 @@
     "hardhat": "internal/cli/cli.js"
   },
   "engines": {
-    "node": "^12.0.0 || ^14.0.0 || ^16.0.0"
+    "node": "^14.0.0 || ^16.0.0 || ^18.0.0"
```

```suggestion
    "node": "^12.0.0 || ^14.0.0 || ^16.0.0 || ^18.0.0"
```

This is unfortunate, but let's play nicely here.